### DISCUSSION TITLE
Unable to log/capture pytest header in a file

### DISCUSSION THREAD

**Discussion by tilusnet** [pytest-dev/pytest#10050] [General]
pytest is nicely outputting a header like 
```
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.9.13, pytest-7.1.2, pluggy-1.0.0
rootdir: ..., configfile: pytest.ini
plugins: timeout-2.1.0, cov-2.12.1

<header ends>
collected 5 items 
....

```

I find it impossible to get this header into a file. 

`pytest 2>&1 > my_file.txt` doesn't work either.
Why?

**Comment by nicoddemus** [comment]
Hi @tilusnet,

Can't reproduce this on Windows:

```
λ pytest .tmp\test_foo.py 2>&1 > out.txt
λ cat out.txt
======================== test session starts ========================
platform win32 -- Python 3.9.8, pytest-7.2.0.dev91+g1689b3886.d20220613, pluggy-1.0.0
rootdir: e:\projects\pytest\.tmp, configfile: pytest.ini
plugins: hypothesis-6.34.1, flake8-1.0.7, flakes-4.0.5
collected 1 item

.tmp\test_foo.py .                                             [100%]

========================= 1 passed in 0.00s =========================
```

Also not sure why this wouldn't work on Linux... 🤔

**Reply by tilusnet** [reply]
It seems, it has to do with a `pytest.ini` present. 
But what affects it?:

```
log_cli = 1
log_cli_level = INFO

markers =
    integration: mark a test as an integration test
    snapshot: mark a test as a snapshot test
log_level = WARNING
```

**Reply by tilusnet** [reply]
Where on earth is the header written if neither `stdout` nor `stderr` can capture it?
I see the header on the screen!

**Reply by nicoddemus** [reply]
I get the same results even with a `pytest.ini` with those contents... definitely something weird going on.

Can you go to another directory (outside the project you are working on), create a new virtual environment, install pytest, run the command, and check if it gets redirected to a file? I suggest that to try to isolate if this is related to your current environment or the test suite you are trying to execute.
